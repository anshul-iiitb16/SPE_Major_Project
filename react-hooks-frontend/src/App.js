import './App.css';
import {BrowserRouter as Router,Routes, Route} from 'react-router-dom'
import FooterComponent from './components/FooterComponent';
import HeaderComponent from './components/HeaderComponent';
import ListEmployeeComponent from './components/ListEmployeeComponent';
import AddEmployeeComponent from './components/AddEmployeeComponent';
import LoginComponent from './components/LoginComponent';


function App() {
  return (
    <div>
      <Router>
        <HeaderComponent />
        <div className='container'>
          <Routes>
            <Route path='/' Component= {LoginComponent}></Route>
            <Route path='*' Component= {ListEmployeeComponent}></Route
            <Route path='/employees' Component= {ListEmployeeComponent}></Route>
            <Route path='/add-employee' Component= {AddEmployeeComponent}></Route>
            <Route path='/edit-employee/:id' Component= {AddEmployeeComponent}></Route>
          </Routes>
        </div>
        <FooterComponent />
      </Router>
    </div>
  );
}

export default App;
