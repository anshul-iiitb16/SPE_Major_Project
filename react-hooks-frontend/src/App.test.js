// import { render, screen } from '@testing-library/react';
// import App from './App';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import App from './App';

it("Renders the App correctly", () => {
  const tree = TestRenderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
