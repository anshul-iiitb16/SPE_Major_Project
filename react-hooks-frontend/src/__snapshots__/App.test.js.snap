// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the App correctly 1`] = `
<div>
  <div>
    <header>
      <nav
        className="navbar navbar-expand-md navbar-dark bg-dark"
      >
        <div>
          <a
            className="navbar-brand"
            href="/"
          >
            Employee Management Application
          </a>
        </div>
      </nav>
    </header>
  </div>
  <div
    className="container"
  >
    <div
      className="container mt-5"
    >
      <h2
        className="text-center"
      >
        Login
      </h2>
      <div
        className="card col-md-6 offset-md-3"
      >
        <div
          className="card-body"
        >
          <form>
            <div
              className="mb-3"
            >
              <label
                className="form-label"
                htmlFor="username"
              >
                Username:
              </label>
              <input
                className="form-control"
                id="username"
                onChange={[Function]}
                placeholder="Enter username"
                type="text"
                value=""
              />
            </div>
            <div
              className="mb-3"
            >
              <label
                className="form-label"
                htmlFor="password"
              >
                Password:
              </label>
              <input
                className="form-control"
                id="password"
                onChange={[Function]}
                placeholder="Enter password"
                type="password"
                value=""
              />
            </div>
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              Login
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div>
    <footer
      className="footer"
    >
      <span
        className="text-muted>"
      >
        All Rights reserved
      </span>
    </footer>
  </div>
</div>
`;
